src/components/slider/slider.tsx

```diff
@@ -0,0 +1,602 @@
+        }
+        return 100 * step / (max - min);
+    }
+
+    private getRelativeValue(value: number, min: number, max: number, step?: Step): number {
+        const normilizedMax = max - min;
+        const normilizedValue = value - min;
+
+        const relativeValue = (normilizedValue * 100) / normilizedMax;
+
+        return this.getValueInRange(relativeValue, 0, 100);
+    }
+
+    private getAbsoluteValue(relativeValue: number): number {
+        const range = this.props.max! - this.props.min!;
+        const absoluteValue = range * relativeValue / 100 + this.props.min!;
+        return this.getValueInRange(absoluteValue, this.props.min!, this.props.max!);
+    }
+
+    private getValueInRange(value: number, min: number, max: number): number {
+        return value < min ? min : (value > max ? max : value);
+    }
+
+    private getValueFromElementAndPointer(element: HTMLElement, pointerPosition: PointerPosition): number {
+        const {relativeStep, isVertical, isReverse} = this.state;
+
+        const sliderBounds = element.getBoundingClientRect();
+
+        const sliderOffset = isVertical ? sliderBounds.top : sliderBounds.left;
+        const sliderSize = isVertical ? sliderBounds.height : sliderBounds.width;
+        const sliderCoordinate = isVertical ? pointerPosition.clientY : pointerPosition.clientX;
+
+        let relativeValue = this.getRelativeValue(sliderCoordinate - sliderOffset, 0, sliderSize);
+
+        relativeValue = isReverse ?
+            (isVertical ? relativeValue : 100 - relativeValue) :
+            (isVertical ? 100 - relativeValue : relativeValue);
+
+        if (typeof relativeStep === 'undefined' || relativeStep === CONTINUOUS_STEP) {
+            return relativeValue;
+        }
+        let value = Math.round(relativeValue / relativeStep) * relativeStep;
+        value = value > 100 ?
+            value - relativeStep :
+            (value < 0 ? value + relativeStep : value);
+
+        return value;
+    }
+
+    private onSliderAreaMouseDown = (event: React.MouseEvent<HTMLElement>) => {
+        if (this.props.disabled) {
+            return;
+        }
+
+        this.props.environment!.addEventListener('mousemove', this.onSliderAreaMouseMove);
+        this.props.environment!.addEventListener('mouseup', this.onSliderAreaMouseUp);
+
+        event.preventDefault();
+        this.focusableElement.focus();
+
+        const relativeValue = this.getValueFromElementAndPointer(this.sliderArea, event);
+
+        this.setState({
+            relativeValue,
+            isActive: true
+        });
+        this.isActive = true;
+
+        this.onDragStart(event.nativeEvent);
+        this.props.onInput!(String(this.getAbsoluteValue(relativeValue)));
+    }
+
+    private onSliderAreaMouseMove = (event: MouseEvent) => {
+        const relativeValue = this.getValueFromElementAndPointer(this.sliderArea, event);
+
+        requestAnimationFrame(() => {
+            if (!this.isSliderMounted) {
```

Comment:
you can use `cancelAnimationFrame` in `componentWillUnmount` to avoid this check